src/platform/backends/hyperv_api/hcn/hyperv_hcn_ipam.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#pragma once
+
+#include <hyperv_api/hcn/hyperv_hcn_ipam_type.h>
+#include <hyperv_api/hcn/hyperv_hcn_subnet.h>
+
+#include <vector>
+
+namespace multipass::hyperv::hcn
+{
+
+struct HcnIpam
```

Comment:
Here is one place where I wouldn't have minded some docs letting me know what an IPAM means.